File: server/pkg/api/world_overviews.go
package api

const (
	WorldOverviewsQueryTypeBySystemID = "bySystemId"
	// WorldOverviewsQueryTypeByResources = "byResources"
)

type WorldOverviewsQueryBySystemID struct {
	SystemID string `json:"systemId"`
}

type WorldOverviewsTableRow struct {
	WorldID string `json:"worldId"`

	MassEarths float64 `json:"massEarths,omitempty"`
	RadiusKm   float64 `json:"radiusKm,omitempty"`
	AgeByrs    float64 `json:"ageByrs,omitempty"`
	Class      string  `json:"class,omitempty"`

	AxisTiltRads      float64 `json:"axisTiltRads,omitempty"`
	DayLengthGameDays float64 `json:"dayLengthGD,omitempty"`

	Size       int  `json:"size,omitempty"`
	IsExplored bool `json:"isExplored"`

	AvgTempK    float64 `json:"avgTempK"`
	PressureBar float64 `json:"surfacePressureBar"`
	GravityGs   float64 `json:"g"`
}

File: ui/src/store/datafront/world_overviews.ts
import { WorldClass, type WorldOverview } from '@/domain/WorldOverview';
import { createDatafrontTable } from '@/lib/datafront/table';
import { type WorldOverviewsQueryBySystemID, type WorldOverviewsTableRow } from '@/lib/net/types.generated';
import { ws } from '@/lib/net/ws';
import { updater, cleaner } from './misc';

export const dfWorldOverviews = createDatafrontTable<WorldOverviewsTableRow, WorldOverview>({
    name: 'world_overviews',
    ws,
    updater,
    cleaner,
    map: (data) => {
        return {
            id: data.worldId,
            params: {
                ageByrs: data.ageByrs ?? 0,
                class: WorldClass.fromString(data.class ?? ''),
                radiusKm: data.radiusKm ?? 0,
                massEarths: data.massEarths ?? 0,
                axisTilt: data.axisTiltRads ?? 0,
                dayLength: data.dayLengthGD ?? 0,
            },
            isExplored: data.isExplored,
            size: data.size ?? 0,
            surface: {
                tempK: data.avgTempK,
                pressureBar: data.surfacePressureBar,
                g: data.g,
            },
            population: {
                bases: data.nBases,
                cities: data.nCities,
                pops: data.nPops,
            },
        };
    },
});

export const dfWorldOverviewsBySystemId = dfWorldOverviews.createQuery<WorldOverviewsQueryBySystemID>(
    'world_overviews/bySystemId',
    (p) => p.systemId,
);
